cli/src/pcluster/models/imagebuilder.py

```diff
@@ -12,160 +12,172 @@
-
-    def _validate(self):
-        self._execute_validator(
-            EbsVolumeTypeSizeValidator,
-            volume_type=ROOT_VOLUME_TYPE,
-            volume_size=self.image.root_volume.size,
-        )
-        self._execute_validator(
-            AMIVolumeSizeValidator,
-            volume_size=self.image.root_volume.size,
-            image=self.build.parent_image,
-        )
+import copy
+import json
+import logging
+
+from common.aws.aws_api import AWSApi
+from common.aws.aws_resources import Stack
+from common.boto3.common import AWSClientError
+from pcluster.models.common import BaseTag
+from pcluster.schemas.imagebuilder_schema import ImageBuilderSchema
+from pcluster.templates.cdk_builder import CDKTemplateBuilder
+from pcluster.utils import get_installed_version
+
+ImageBuilderStatusMapping = {
+    "BUILD_IN_PROGRESS": [
+        "CREATE_IN_PROGRESS",
+        "ROLLBACK_IN_PROGRESS",
+        "UPDATE_IN_PROGRESS",
+        "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS",
+        "UPDATE_ROLLBACK_IN_PROGRESS",
+        "UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS",
+        "REVIEW_IN_PROGRESS",
+        "IMPORT_IN_PROGRESS",
+        "IMPORT_ROLLBACK_IN_PROGRESS",
+    ],
+    "BUILD_FAILED": [
+        "CREATE_FAILED",
+        "ROLLBACK_FAILED",
+        "ROLLBACK_COMPLETE",
+        "UPDATE_ROLLBACK_FAILED",
+        "UPDATE_ROLLBACK_COMPLETE",
+        "IMPORT_ROLLBACK_FAILED",
+        "IMPORT_ROLLBACK_COMPLETE",
+    ],
+    "BUILD_COMPLETE": ["CREATE_COMPLETE", "UPDATE_COMPLETE", "IMPORT_COMPLETE"],
+    "DELETE_IN_PROGRESS": ["DELETE_IN_PROGRESS"],
+    "DELETE_FAILED": ["DELETE_FAILED"],
+    "DELETE_COMPLETE": ["DELETE_COMPLETE"],
+}
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ImageBuilderActionError(Exception):
+    """Represent an error during the execution of an action on the imagebuilder."""
+
+    def __init__(self, message: str, validation_failures: list = None):
+        super().__init__(message)
+        self.validation_failures = validation_failures or []
+
+
+class ImageBuilderStack(Stack):
+    """Class representing a running stack associated to a building image."""
+
+    def __init__(self, stack_data: dict):
+        """Init stack info."""
+        super().__init__(stack_data)
+        try:
+            self._image_resource = AWSApi().instance().cfn.describe_stack_resouce(self.name, "ParallelClusterImage")
+        except AWSClientError:
+            self._image_resource = None
+
+    @property
+    def version(self):
+        """Return the version of ParallelCluster used to create the stack."""
+        return self._get_tag("pcluster_version")
+
+    @property
+    def image_id(self):
+        """Return output image id."""
+        if self._image_resource:
+            try:
+                image_build_version_arn = self._image_resource.get("StackResourceDetail").get("PhysicalResourceId")
+                image_id = AWSApi.instance().imagebuilder.get_image_id(image_build_version_arn)
```

Comment:
minor: you can avoid defining the image_id

Reply:
done